lib/reversetunnel/conn.go

```diff
@@ -251,3 +254,77 @@ func (c *remoteConn) sendDiscoveryRequest(req discoveryRequest) error {
 
 	return nil
 }
+
+func min(a, b int) int {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+const emitConnTargetPrefix = "Teleport"
+
+// emitConn is a wrapper for a net.Conn that emits an audit event for non-Teleport connections.
+type emitConn struct {
+	net.Conn
+	mu       *sync.RWMutex
+	buffer   bytes.Buffer
+	emitter  apievents.Emitter
+	ctx      context.Context
+	serverID string
+	emitted  bool
+}
+
+func newEmitConn(ctx context.Context, conn net.Conn, emitter apievents.Emitter, serverID string) *emitConn {
+	return &emitConn{
+		Conn:     conn,
+		mu:       &sync.RWMutex{},
+		buffer:   bytes.Buffer{},
+		emitter:  emitter,
+		ctx:      ctx,
+		serverID: serverID,
+		emitted:  false,
+	}
```

Comment:
nit: drop zero types.

```suggestion
	return &emitConn{
		Conn:     conn,
		ctx:      ctx,
		emitter:  emitter,
		serverID: serverID,
	}
```

(Assuming you took the non-pointer-mutex comment above.)